gstreamer-sdp/src/s_d_p_message.rs

```diff
@@ -1,11 +1,617 @@
+        };
+        match result {
+            SDPResult::ok => Some(_),
+            _ => Err(()),
+            }
+        }
+    }
+
+    pub fn add_media(&mut self, media: SDPMedia) -> Result<_, ()> {
+        let result = unsafe {
+            from_glib(ffi::gst_sdp_message_add_media(self.to_glib_none_mut().0, media.to_glib_full()))
+        };
+        mem::forget(media);
+        match result {
+            SDPResult::ok => Some(_),
+            _ => Err(()),
+            }
+        }
+    }
+
+    pub fn add_phone(&mut self, phone: &str) -> Result<_, ()> {
+        let result = unsafe {
+            from_glib(ffi::gst_sdp_message_add_phone(self.to_glib_none_mut().0, phone.to_glib_none().0))
+        };
+        match result {
+            SDPResult::ok => Some(_),
+            _ => Err(()),
+            }
+        }
+    }
+
+    pub fn add_time(&mut self, start: &str, stop: &str, repeat: &[&str]) -> Result<_, ()> {
+        let result = unsafe {
+            from_glib(ffi::gst_sdp_message_add_time(self.to_glib_none_mut().0, start.to_glib_none().0, stop.to_glib_none().0, repeat.to_glib_none().0))
+        };
+        match result {
+            SDPResult::ok => Some(_),
+            _ => Err(()),
+            }
+        }
+    }
+
+    pub fn add_zone(&mut self, adj_time: &str, typed_time: &str) -> Result<_, ()> {
+        let result = unsafe {
+            from_glib(ffi::gst_sdp_message_add_zone(self.to_glib_none_mut().0, adj_time.to_glib_none().0, typed_time.to_glib_none().0))
+        };
+        match result {
+            SDPResult::ok => Some(_),
+            _ => Err(()),
+            }
+        }
+    }
+
+    pub fn as_text(&self) -> Option<String> {
+        unsafe {
+            from_glib_full(ffi::gst_sdp_message_as_text(self.to_glib_none().0))
+        }
+    }
+
+    pub fn attributes_len(&self) -> u32 {
+        unsafe {
+            ffi::gst_sdp_message_attributes_len(self.to_glib_none().0)
+        }
+    }
+
+    pub fn attributes_to_caps(&self, caps: &mut gst::CapsRef) -> Result<_, ()> {
+        let result = unsafe {
+            from_glib(ffi::gst_sdp_message_attributes_to_caps(self.to_glib_none().0, caps.to_glib_none_mut().0))
+        };
+        match result {
+            SDPResult::ok => Some(_),
+            _ => Err(()),
+            }
+        }
+    }
+
+    pub fn bandwidths_len(&self) -> u32 {
+        unsafe {
+            ffi::gst_sdp_message_bandwidths_len(self.to_glib_none().0)
+        }
+    }
+
+    pub fn dump(&self) -> Result<_, ()> {
+        let result = unsafe {
+            from_glib(ffi::gst_sdp_message_dump(self.to_glib_none().0))
+        };
+        match result {
+            SDPResult::ok => Some(_),
+            _ => Err(()),
+            }
+        }
+    }
+
+    pub fn emails_len(&self) -> u32 {
+        unsafe {
+            ffi::gst_sdp_message_emails_len(self.to_glib_none().0)
+        }
+    }
+
+    pub fn get_attribute(&self, idx: u32) -> Option<SDPAttribute> {
```

Comment:
Borrowing for all getters. You can do this by marking the struct as `#[repr(C)]` and then doing `&*(the_ptr as *mut SDPAttribute)`, i.e. casting from the FFI type to your wrapper struct